src/pages/workspaces/workspace/Data.js

```diff
@@ -275,6 +284,9 @@ class WorkspaceDataContent extends Component {
               ref: this.table,
               width, height,
               rowCount: entities.length,
+              onScroll: saveScroll,
+              initialX: StateHistory.get().initialX,
+              initialY: StateHistory.get().initialY,
```

Comment:
These should be extracted, probably in the constructor, so we don't fetch history on every render.